### ISSUE TITLE
false posivite on float_cmp when comparing between f32 pointer

### ISSUE THREAD

**Issue by bungcip** [Manishearth/rust-clippy#1037]
this rust snippet
```rust
fn main() {
    let xs : [f32; 1] = [0.0];
    let a: *const f32 = xs.as_ptr();
    let b: *const f32 = xs.as_ptr();

    assert!(a == b);
}
```
give false positive lint 
```
src\main.rs:6:13: 6:19 warning: ==-comparison of f32 or f64 detected. Consider changing this to `(a - b).abs() < epsilon` for some suitable value of epsilon. std::f32::EPSILON and std::f64::EPSILON are available., #[warn(float_cmp)] on by default
src\main.rs:6     assert!(a == b);
```
